### ISSUE TITLE
No such file or directory [~arma/AoW] / Invalid env format - Ubuntu 20.10/Flatpak

### ISSUE THREAD

**Issue by Jackson-Bondy** [muttleyxd/arma3-unix-launcher#152]
Fresh install of Ubuntu 20.10 and Steam/Proton/Arma.
Steam is the Flatpak version and Arma is installed on a secondary drive.

Launcher opens with the following in console:

```
arma3-unix-launcher -v
conf file: "/home/jackson/.config/a3unixlauncher/launcher.conf"
Steam initialization failed - Flatpak detected!
Error loading mod from config:
---
{
  "enabled": false,
  "name": "cannot read name",
  "path": "~arma/AoW"
}
--- Reason: filesystem error: directory iterator cannot open directory: No such file or directory [~arma/AoW]
```

There is a mod in the launcher called "cannot read name" with the path `~arma/AoW`

![image](https://user-images.githubusercontent.com/79606580/109066132-828f9100-76e4-11eb-8d76-a89f5fe78cb3.png)

Clicking Start outputs `error: Invalid env format`

Config file:
```

    "mods": [
        {
            "enabled": false,
            "name": "cannot read name",
            "path": "~arma/AoW"
        }
    ],
    "parameters": {
        "checkSignatures": false,
        "connect": null,
        "cpuCount": -1,
        "customParameters": null,
        "dlcContact": true,
        "dlcGlobalMobilization": false,
        "enableHT": false,
        "environmentVariables": null,
        "exThreads": -1,
        "filePatching": false,
        "host": false,
        "hugepages": false,
        "name": null,
        "noLogs": false,
        "noPause": false,
        "noSplash": false,
        "par": null,
        "password": null,
        "port": null,
        "protonDisableEsync": false,
        "skipIntro": false,
        "window": false,
        "world": null
    },
    "paths": {
        "arma": "/media/jackson/Games/Steam/steamapps/common/Arma 3",
        "workshop": "/media/jackson/Games/Steam/steamapps/workshop/content/107410"
    },
    "settings": {
        "theme": "System"
    }
}
```

Fixes tried:
Compiling and installing from source
Installing from .debian
Using AppImage

Reinstalling Arma
Moving Arma to the boot drive

Creating the directory `~arma/AoW` in the Workshop folder

Unsubscribing from all Workshop mods.

Not sure what else to try at this point, I'm returning to Linux after a long time away and I couldn't find anything about any of the errors I was having online so I suspect it's something I've done.

**Comment by muttleyxd** [comment]
Closed by #153

Feel free to reopen if issue persists.

**Comment by CryptArma** [comment]
Hi, sorry for my english, I am using Manjaro 20.2.1 with Steam/Proton in flatpak and apparently the error "Invalid env format" is being caused by "environmentVariables" in the launcher config (/home/user/.config/a3unixlauncher/launcher.conf) having "null" as the value when running for the first time but when you change the value for something else like "false" the launcher will spit this in the terminal and it will work:

conf file: "/home/user/.config/a3unixlauncher/launcher.conf"
Steam initialization failed - Flatpak detected!
exception while parsing settings["environmentVariables"]:
[json.exception.type_error.302] type must be string, but is boolean
INFO:root:https://github.com/flathub/com.valvesoftware.Steam/wiki
INFO:root:Will set XDG dirs prefix to /home/crypt/.var/app/com.valvesoftware.Steam
INFO:root:Overriding TZ to Location_of_the_user
Running Steam on org.freedesktop.platform 20.08.6 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Steam client's requirements are satisfied
/home/user/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/steam -applaunch 107410 -nolauncher -environmentVariables= -skipIntro

Then after that the value of "environmentVariables" will be changed to (" ") automatically. Thank you for you patience and hard work.
